### DISCUSSION TITLE
Tests running time

### DISCUSSION THREAD

**Discussion by SoulPancake** [redis/rueidis#582] [Ideas]
@rueian I think the overall testsuite take a very long to run, can we look into optimizing this somehow?
Also having a huge chunk straight up in the rueidis root package can also be modularised somehow I think? 
What are your thoughts on this?

**Comment by rueian** [comment]
Hi @SoulPancake, most time-consuming tests can be ignored by the `-short` flag. For example, this is what I do mostly on my machine:

```sh
▶ go test -short -coverprofile=coverage.out -race github.com/redis/rueidis
ok  	github.com/redis/rueidis	88.099s	coverage: 99.3% of statements
```

It takes about 90 seconds to run all the short tests.

What about yours? Which part slows you down?

**Comment by SoulPancake** [comment]
Using -short flag it is relatively faster
Comparable to your time
Thanks!
I was just wondering more about the actual tests that run on CircleCI after we push our code, Do you not think it takes a while? Can we further optimise it somehow? @rueian
